### DISCUSSION TITLE
v3 features suggestions

### DISCUSSION THREAD

**Discussion by sh-dv** [sh-dv/hat.sh#73] [Ideas]
Hello everybody!
I am excited to announce that i will start the development of v3 of hat.sh in the near future.
The estimated release date of the `beta` will be in late December this year.

Many new features will be in the next version including :
- File hashing
- Verifying signatures
- Encryption of private keys
- Keyfile encryption
- File name encryption
- Adding comments to encrypted file
- Directory Zipping
- Splitting encrypted files into chunks
- Passphrases generator
- And many more...

Please let me know in the comments about new features you would like to see in the next version.
Regards, sh-dv.

**Comment by covert-encryption** [comment]
Implement Covert format :)

**Comment by valsha** [comment]
Hi, what is ETA v3? Thanks.

**Comment by hakavlad** [comment]
Suggestions:
- Stronger KDF parameters
- Randomized padding or even
- [PURB](https://en.wikipedia.org/wiki/PURB_(cryptography)) format
